dali/test/python/test_operator_get_property.py

```diff
@@ -0,0 +1,156 @@
+                source_info) == f"archive {root_path}tar file at \"{root_path}\"component offset {offset}"
+
+
+def test_wds_properties():
+    for dev in ['cpu', 'gpu']:
+        yield _test_wds_properties, dev
+
+
+@pipeline_def
+def tfr_properties(root_path, index_path, device):
+    import nvidia.dali.tfrecord as tfrec
+    inputs = fn.readers.tfrecord(path=root_path, index_path=index_path,
+                                 features={"image/encoded": tfrec.FixedLenFeature((), tfrec.string, ""),
+                                           "image/class/label": tfrec.FixedLenFeature([1], tfrec.int64, -1)})
+    enc, lab = fn.get_property(inputs["image/encoded"], key="source_info"), \
+               fn.get_property(inputs["image/class/label"], key="source_info")
+    if device == 'gpu':
+        enc = enc.gpu()
+        lab = lab.gpu()
+    return enc, lab
+
+
+def _test_tfr_properties(device):
+    root_path = os.path.join(get_dali_extra_path(), 'db', 'tfrecord', 'train')
+    index_path = os.path.join(get_dali_extra_path(), 'db', 'tfrecord', 'train.idx')
+    idx = [0, 171504, 553687, 651500, 820966, 1142396, 1380096, 1532947]
+    p = tfr_properties(root_path, index_path, device, batch_size=8, num_threads=4, device_id=0)
+    p.build()
+    output = p.run()
+    for out in output:
+        out = out if device == 'cpu' else out.as_cpu()
+        for source_info, ref_idx in zip(out, idx):
+            assert _uint8_tensor_to_string(source_info) == f"{root_path} at index {ref_idx}"
+
+
+def test_tfr_properties():
+    for dev in ['cpu', 'gpu']:
+        yield _test_tfr_properties, dev
+
+
+@pipeline_def
+def es_properties(layouts, device):
+    num_outputs = len(layouts)
+
+    def gen_data():
+        yield np.random.rand(num_outputs, 3, 4, 5)
+
+    inp = fn.external_source(source=gen_data, layout=layouts, num_outputs=num_outputs, batch=False, cycle=True,
+                             device=device)
+    return tuple(fn.get_property(i, key="layout") for i in inp)
+
+
+def _test_es_properties(device):
+    layouts = ['ABC', 'XYZ']
+    p = es_properties(layouts, device, batch_size=8, num_threads=4, device_id=0)
+    p.build()
+    output = p.run()
+    for out, lt in zip(output, layouts):
+        out = out if device == 'cpu' else out.as_cpu()
+        for sample in out:
+            assert _uint8_tensor_to_string(sample), lt
+
+
+def test_es_properties():
+    for dev in ['cpu', 'gpu']:
+        yield _test_es_properties, dev
+
+
+@pipeline_def
+def improper_property(root_path, device):
+    read = fn.readers.webdataset(paths=[root_path], ext=['jpg'])
+    return fn.get_property(read, key=["this key doesn't exist"])
+
+
+def _test_improper_property(device):
+    root_path = os.path.join(get_dali_extra_path(), "db/webdataset/MNIST/devel-0.tar")
+    p = improper_property(root_path, device, batch_size=8, num_threads=4, device_id=0)
+    p.build()
+    output = p.run()
+
+
+@raises
+def test_improper_property():
+    for dev in ['cpu', 'gpu']:
```

Comment:
Please use `assert_raises` from DALI nose_utils.py and check the exact message and error type.

Reply:
Done